src/cls/journal/cls_journal.cc

```diff
@@ -540,10 +540,12 @@ int journal_client_register(cls_method_context_t hctx, bufferlist *in,
                             bufferlist *out) {
   std::string id;
   bufferlist data;
+  std::set<cls::journal::Client> clients;
   try {
     bufferlist::iterator iter = in->begin();
     ::decode(id, iter);
     ::decode(data, iter);
+    ::decode(clients, iter);
```

Comment:
@dillaman With the updated patch, I still call `journal_client_list()` from `journal_client_register()` and have helper methods to decode and aggregate clients (with batching).

Also, these helpers reside in cls_journal_client.cc but needs to be called from cls_journal.cc vai externs.